meetings/2017-02-08.md

```diff
@@ -0,0 +1,211 @@
+
+@rvagg: See also
+https://github.com/nodejs/node/blob/master/tools/license-builder.sh.
+
+James will audit all files and take needed action.
+
+**Next steps:**
+
+* James to continue work on
+  [node#10599](https://github.com/nodejs/node/pull/10599).
+* Keep on agenda for next week to review Board’s action.
+
+---
+
+### ES Modules: Module Loader [node-eps#39](https://github.com/nodejs/node-eps/pull/39)
+
+Async or sync for `require()`? Support mixed ES and CJS trees in `require()`?
+
+@MylesBorins has talked with many in Google and other stakeholders, sentiment is
+that some compromise must be made, and best to keep difference between Node.js
+and browsers as small as possible. So `import` keyword should be asynchronous. 
+
+In V8 loader, path resolution of modules is asynchronous, but evaluation of
+modules is deterministic and same as synchronous `require()`.
+
+Could still have a synchronous loader through another function, e.g. some
+evolution of `require(...)`.
+
+@jasnell: Cannot synchronously `require(‘esm’)`, and probably cannot
+asynchronously `import cjs`.
+
+@mylesborins: `import` keyword must be top of file, `import()` function can be
+used later. `import()` function can be treated as synchronous by `await
+import()`.
+
+@Fishrock123: still problematic for deps tree that includes ES and CJS modules
+[Editor: because of Temporal Dead Zone].
+
+@jasnell: and even more complex if there are circular deps.
+
+@jasnell: Easiest thing to do is use asynchronous `import/import()` for ES and
+synchronous `require` for CJS.
+
+Next step should be to begin implementation and adjust if needed.
+
+@Fishrock123: If we want a sync loader we need a second type of ModuleRecord (in
+ECMA-262) to avoid the Temporal Dead Zone between parse and eval time for ES
+modules.
+
+@rvagg: We need to guard Node’s interests even if they’re not the same as
+browsers. But in this case not necessarily in Node’s interest to go sync.
+
+@mylesborins: Many JS developers work in both browser and non-browser
+environments, and compatibility between them is one of the reasons for the
+success of Node.
+
+@evanlucas: you can’t exactly jump between them now, e.g. `fs` is only in Node.
+
+@jasnell: We try to differ only when we have to, like `fs`. If it’s reasonable
+to match browsers with an async loader we should do that.
+
+Those who have opinions on CTC and @bmeck are leaning toward async.
+
+Begin implementation and learn and adjust as needed. Begin with async since if
+successful it brings us closer to browsers; if unsuccessful fall back to the
+easier path with a sync loader.
+
+@mylesborins suggested a meeting with CTC and TC39 in May:
+[CTC#67](https://github.com/nodejs/CTC/issues/67).
+
+@Trott will open an issue specifically to track implementation of an async
+module loader and keep that on ctc-agenda.
+
+**Next steps:**
+
+* Begin implementation of an async loader and learn and adjust as needed.
+* @Trott will open an issue to track.
```

Comment:
https://github.com/nodejs/node/issues/11233 although I wonder if it would have been better to open it in CTC repo. More visibility there though, so ¯\\_(ツ)_/¯